### PR TITLE
Refactor FXIOS-10657 connectETPSettingsNavigationWithBrowserVC 

### PR DESCRIPTION
**Pull request by petruSt24**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10657)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/23317)

## :bulb: Description

Connected the nav to settings with the BVC nav handling
## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by petruSt24** [comment]
@cyndichin tried another approach, hopefully I understood your vision this time

**Review by cyndichin — APPROVED** [review]
thanks @petruSt24! this is more aligned with what I was thinking! looks good. Ideally, I would want to consolidate how we handle showing settings view, but we can do this at a later date. approved!

File: firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift [hunk 1]
```diff
@@ -190,7 +190,18 @@ class TrackingProtectionViewController: UIViewController,
...
         trackingProtectionState = state
-        // TODO: FXIOS-10657 connect the ETP navigation with the BVC general navigation
+        if let navDestination = state.navigateTo {
+            switch navDestination {
+            case .home:
+                navigationController?.popToRootViewController(animated: true)
+            case .back:
+                navigationController?.popViewController(animated: true)
+            case .close:
+                enhancedTrackingProtectionMenuDelegate?.didFinish()
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:200]: these are simple calls so okay with leaving it here, but curious why not add the navigation logic to the coordinator instead?

Comment (petruSt24) [reply firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift:200]: I'm not a big fan of one liner methods called once

**Pull request merged**